### PR TITLE
data transfer hash(issue 1269)

### PR DESCRIPTION
**Pull request by x418-22n**
finished the design pattern data transfer hash(issue 1269)

### CODE REVIEW

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Presentation.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        hash.create(k, v);
+        
+    }
+    public Object get(String k,Hash hash)
+    {
+        
+        return  hash.get(k);
+        
+    }
+
```
Comment (ohbus) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Presentation.java:16]: please remove the extra line spaces

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+public class Business {
+    public void create(String k,Object v,Hash hash)
+    {
+        hash.create(k, v);
+        
+    }
+    public Object get(String k,Hash hash)
+    {
+        
+
```
Comment (ohbus) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java:13]: remove extra spaces

**Review by ohbus — CHANGES_REQUESTED** [review]
You have added the bytecode instead of the actual code.

Please submit the java code re-evaluation.

**Review by ohbus — CHANGES_REQUESTED** [review]
I wonder why the bytecode is still here and how it managed to get inside git tracking instead of being mentioned in the gitignore 😕 

1. Please take a look into these other patterns for understanding how you would like to design the readme section of the patterns from the examples below besides following the [**Wiki**](https://github.com/iluwatar/java-design-patterns/wiki) section.
    - https://github.com/iluwatar/java-design-patterns/tree/master/abstract-factory
    - https://github.com/iluwatar/java-design-patterns/tree/master/active-object
    - https://github.com/iluwatar/java-design-patterns/tree/master/acyclic-visitor
    - https://github.com/iluwatar/java-design-patterns/tree/master/acyclic-visitor
    - https://github.com/iluwatar/java-design-patterns/tree/master/aggregator-microservices
    - https://github.com/iluwatar/java-design-patterns/tree/master/ambassador
    - https://github.com/iluwatar/java-design-patterns/tree/master/api-gateway
    - https://github.com/iluwatar/java-design-patterns/tree/master/factory-method
    - https://github.com/iluwatar/java-design-patterns/tree/master/flyweight
    - https://github.com/iluwatar/java-design-patterns/tree/master/game-loop
    - https://github.com/iluwatar/java-design-patterns/tree/master/iterator

2. Also add the pattern to the parent pom for it to get executed and perform necessary analysis on your newly submitted code.

If you have any queries or facing difficulties please reach out to me after you have read the [**Wiki**](https://github.com/iluwatar/java-design-patterns/wiki) section properly

File: Data_Transfer_Hash/pom.xml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iluwatar) [inline Data_Transfer_Hash/pom.xml:1]: Folder naming uses snake case. Data_Transfer_Hash -> data_transfer_hash

File: Data_Transfer_Hash/pom.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
```
Comment (iluwatar) [inline Data_Transfer_Hash/pom.xml:28]: I think we can remove this part

File: Data_Transfer_Hash/pom.xml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        <plugin>
+          <artifactId>maven-site-plugin</artifactId>
+          <version>3.7.1</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-project-info-reports-plugin</artifactId>
+          <version>3.0.0</version>
+        </plugin>
+      </plugins>
+    </pluginManagement>
```
Comment (iluwatar) [inline Data_Transfer_Hash/pom.xml:73]: Many of the plugins are already specified in the parent pom.xml. Check how the other pattern pom.xmls have been defined e.g. https://github.com/iluwatar/java-design-patterns/blob/master/abstract-factory/pom.xml

File: Data_Transfer_Hash/readme.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+layout: pattern
+title: data transfer hash
+folder: Data_Transfer_Hash
+permalink: /patterns/Data_Transfer_Hash/
+categories: Architectural
+tags:
+ - Performance
```
Comment (iluwatar) [inline Data_Transfer_Hash/readme.md:8]: I would suggest structural and decoupling. See https://github.com/iluwatar/java-design-patterns/wiki/07.-Categories-and-Tags

File: Data_Transfer_Hash/readme.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+---
+layout: pattern
+title: data transfer hash
```
Comment (iluwatar) [inline Data_Transfer_Hash/readme.md:3]:
```suggestion
title: Data Transfer Hash

```

File: Data_Transfer_Hash/readme.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+folder: Data_Transfer_Hash
+permalink: /patterns/Data_Transfer_Hash/
+categories: Architectural
+tags:
+ - Performance
+---
+
+## Also known as
+
+* data transfer hash
```
Comment (iluwatar) [inline Data_Transfer_Hash/readme.md:13]: This is the original pattern name, right? No need to specify it here.

File: Data_Transfer_Hash/readme.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+## Intent
+
+Providing a  class called data transfer hash, in order to send message between other class.
```
Comment (iluwatar) [inline Data_Transfer_Hash/readme.md:17]: The intent should describe the pattern very clearly and concisely. Please try to rephrase.

File: Data_Transfer_Hash/readme.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+## Explanation
+
+
+> Data transfer hashes may be implemented as simple hash tables (or HashMaps). 
+A more robust implementation uses a container object to hold the hash, as well as identifying information, type-safe data retrieval methods, and well-known keys.
+> https://www.oreilly.com/library/view/j2ee-design-patterns/0596004273/re12.html
```
Comment (iluwatar) [inline Data_Transfer_Hash/readme.md:24]: Add the link to another section, see https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package Data_Transfer_Hash;
+/*
+Data transfer hashes may be implemented as simple hash tables (or HashMaps). 
+A more robust implementation uses a container object to hold the hash, as well as identifying information, type-safe data retrieval methods, and well-known keys.
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java:4]: I would first explain what the pattern is, then how this example implements it

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+/*
+Data transfer hashes may be implemented as simple hash tables (or HashMaps). 
+A more robust implementation uses a container object to hold the hash, as well as identifying information, type-safe data retrieval methods, and well-known keys.
+*/
+
+public class App 
+{
+    public static void main( String[] args )
+    {
+        Hash hash= new Hash();
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java:11]: You can use `var` in many occasions, check throughout the example

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+        Business business=new Business();
+        Presentation presentation=new Presentation();
+        business.create("a", "aaa", hash);
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java:14]: Please use some time to come up with a more interesting example than 'a' and 'aaa'

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        System.out.println("Business create "+"aaa");
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/App.java:15]: Use a proper logger, check elsewhere too

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package Data_Transfer_Hash;
+/*Business objects
+Create hashes for sending data, or use values in received hashes.*/
+public class Business {
+    public void create(String k,Object v,Hash hash)
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java:5]: Consider a more descriptive name for the method

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java [hunk 16]
```diff
@@ -0,0 +1,14 @@
+package Data_Transfer_Hash;
+/*Business objects
+Create hashes for sending data, or use values in received hashes.*/
+public class Business {
+    public void create(String k,Object v,Hash hash)
+    {
+        hash.create(k, v);
+        
+    }
+    public Object get(String k,Hash hash)
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Business.java:10]: Consider naming

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Hash.java [hunk 17]
```diff
@@ -0,0 +1,18 @@
+package Data_Transfer_Hash;
+
+import java.util.Hashtable;
+/*Data transfer hash
+Stores data for transport between layers as a set of values associated with well-known keys.*/
+public class Hash {
+    private Hashtable hash;
+    Hash()
+    {hash=new Hashtable();}
+    public void create(String k,Object v)
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Hash.java:10]: Would `put` be a better name?

File: Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Presentation.java [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+public class Presentation {
+    public void create(String k,Object v,Hash hash)
+    {
+        hash.create(k, v);
+        
+    }
+    public Object get(String k,Hash hash)
+    {       
+        return  hash.get(k);        
+    }
```
Comment (iluwatar) [inline Data_Transfer_Hash/src/main/java/Data_Transfer_Hash/Presentation.java:13]: Consider naming

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
@x418-22n please address the review comments

**Comment by iluwatar** [comment]
Closed due to inactivity

**Pull request closed without merging**